### PR TITLE
Face detection method

### PR DESCRIPTION
**Pull request by Abinashi7**
Draft pull request

### CODE REVIEW

File: helloworld/main.py [hunk 1]
```diff
@@ -1,4 +1,62 @@
...
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = r'GVServiceAccountToken.json'
+
+app = Flask(__name__)
+api = Api(app)
+
+# Set this to false when deploying to live application
+app.config['DEBUG'] = True
+
+# Allow cross-origin resource sharing
+cors = CORS(app, resources={r'/api/*': {'origins': '*'}})
```
Comment (quanganhhoang) [inline helloworld/main.py:19]: why do we need this?

File: helloworld/main.py [hunk 2]
```diff
@@ -20,3 +78,8 @@ def hello_http(request):
     else:
         name = 'World'
     return 'Hello {}!'.format(escape(name))
+
+
+if __name__ == '__main__':
+    # change the port number if 4000 doesn't work on your local machine
+    app.run(host='127.0.0.1', port=4000)
```
Comment (quanganhhoang) [inline helloworld/main.py:85]: is this here for local dev or something?

File: helloworld/main.py [hunk 3]
```diff
@@ -1,4 +1,62 @@
-from flask import escape
+import os
+import io
+from flask import escape, request
+from flask_restful import Resource, Api
+from google.cloud import vision
```
Comment (quanganhhoang) [inline helloworld/main.py:5]:
```
Traceback (most recent call last): File "/layers/google.python.pip/pip/bin/functions-framework", line 8, in <module> sys.exit(_cli()) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/click/core.py", line 829, in __call__ return self.main(*args, **kwargs) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/click/core.py", line 782, in main rv = self.invoke(ctx) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/click/core.py", line 1066, in invoke return ctx.invoke(self.callback, **ctx.params) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/click/core.py", line 610, in invoke return callback(*args, **kwargs) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/functions_framework/_cli.py", line 37, in _cli app = create_app(target, source, signature_type) File "/layers/google.python.pip/pip/lib/python3.8/site-packages/functions_framework/__init__.py", line 172, in create_app spec.loader.exec_module(source_module) File "<frozen importlib._bootstrap_external>", line 783, in exec_module File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "/workspace/main.py", line 5, in <module> from google.cloud import vision ImportError: cannot import name 'vision' from 'google.cloud' (unknown location)
```

Is this fixed?

File: helloworld/main.py (hunk 1 above)
Comment (Abinashi7) [reply helloworld/main.py:19]: We do not. It is just there. I'll get rid of everything unnecessary once I know I can deploy a function.

File: helloworld/main.py [hunk 4]
```diff
@@ -1,4 +1,62 @@
...
+    print(vision_client)
+
+    # image = vision.types.Image()
+    image = vision.Image()
+    image.source.image_uri = uri
+
+    response = vision_client.face_detection(image=image)
+    faceAnnotations = response.face_annotations
+
+    likelihood = ('unknown', 'Very unlikely', 'Unlikely',
```
Comment (quanganhhoang) [inline helloworld/main.py:45]: could you please add comments on what this code is doing?

File: helloworld/main.py (hunk 2 above)
Comment (Abinashi7) [reply helloworld/main.py:85]: Yeah, I was not sure how to test my https function locally without first deploying it. So I tried it over the Postman and it worked.

File: helloworld/main.py (hunk 3 above)
Comment (Abinashi7) [reply helloworld/main.py:5]: I did not see this error on my end.

File: helloworld/main.py (hunk 2 above)
Comment (quanganhhoang) [reply helloworld/main.py:85]: that's cool! Might be something @liuhantao9 wants to look into

File: helloworld/main.py (hunk 4 above)
Comment (Abinashi7) [reply helloworld/main.py:45]: I will put in comments and upload it again. But in this function I am just trying to ensure that Google vision is successfully acting upon when it is provided an image via url. Currently likelihood list is there provide labels when printing an output.
Output example:
Detection Confidence:  0.8303185701370239
Angry likelyhood:  Very unlikely
Joy likelyhood:  Very likely
Sorrow likelyhood:  Very unlikely
Surprise likelyhood:  Very unlikely
Headwear likelyhood:  Very unlikely

File: helloworld/main.py (hunk 2 above)
Comment (liuhantao9) [reply helloworld/main.py:85]: Yeah, I am trying to test my function. There is no route, how do you get http reequest to work? @Abinashi7

**Pull request closed without merging**